### ISSUE TITLE
Worddropout in Flert?

### ISSUE THREAD

**Issue by helpmefindaname** [flairNLP/flair#2420]
Hello considering the training code of Flert:

```
from flair.datasets import CONLL_03
from flair.embeddings import TransformerWordEmbeddings
from flair.models import SequenceTagger
from flair.trainers import ModelTrainer
import torch
from torch.optim.lr_scheduler import OneCycleLR

# 1. get the corpus
corpus = CONLL_03()
print(corpus)

# 2. what label do we want to predict?
label_type = 'ner'

# 3. make the label dictionary from the corpus
label_dict = corpus.make_label_dictionary(label_type=label_type)
print(label_dict)

# 4. initialize fine-tuneable transformer embeddings WITH document context
embeddings = TransformerWordEmbeddings(
    model='xlm-roberta-large',
    layers="-1",
    subtoken_pooling="first",
    fine_tune=True,
    use_context=True,
)

# 5. initialize bare-bones sequence tagger (no CRF, no RNN, no reprojection)
tagger = SequenceTagger(
    hidden_size=256,
    embeddings=embeddings,
    tag_dictionary=label_dict,
    tag_type='ner',
    use_crf=False,
    use_rnn=False,
    reproject_embeddings=False,
)

# 6. initialize trainer with AdamW optimizer
trainer = ModelTrainer(tagger, corpus, optimizer=torch.optim.AdamW)

# 7. run training with XLM parameters (20 epochs, small LR, one-cycle learning rate scheduling)
trainer.train('resources/taggers/sota-ner-flert',
              learning_rate=5.0e-6,
              mini_batch_size=4,
              mini_batch_chunk_size=1,  # remove this parameter to speed up computation if you have a big GPU
              max_epochs=20,  # 10 is also good
              scheduler=OneCycleLR,
              embeddings_storage_mode='none',
              weight_decay=0.,
              )
```
and knowing the `SequenceTagger` having per default 0.05 word_dropout.
I was wondering if this was on purpose or if it was overseen.
From my understanding: As the word_dropout is applied after the embeddings and there is no sequential module afterwards, it implies that for 5% of tokens, the bias vectors are the only relevant indicators for prediction.

From my view this can be seen as one of the following two:
a) a kind of regularization, that leads the bias vectors into the direction of the label distribution
b) a waste of computation, as it reduces the gradient information by 5%

If it is b) it might make sense to update the documentation

**Comment by alanakbik** [comment]
Hello @helpmefindaname oh this is a good point! I think we left the word_dropout on 0.05 by default which I think you're right makes little sense here. I'll do an experiment if setting it to 0 improves tagging results.

**Comment by alanakbik** [comment]
Update: setting word_dropout to 0 or even to 0.1 does not seem to change much in the final numbers, at least in my few experimental runs.

**Comment by helpmefindaname** [comment]
Thank you for trying out! I'd assume setting it to 0 would still be the best practice, even if it doesn't really matter, so I'm going to use that in my further experiments.
